client/src/checkbox.js

```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import "./index.css";
+
+function Checkbox({ name, id, index, isChecked, eventHandler }) {
+  return (
+    <div className="checkbox">
+      <label htmlFor={name}>
+        <input
+          type="checkbox"
+          id={name}
+          name={name}
+          value={id}
+          checked={isChecked}
+          onChange={() => eventHandler(index)}

```

Comment:
I tried wrapping each checkbox in a div and putting the handling logic there:
```
{allGenres.map((genre, index) => (
            <div className="clickHandler" onClick={handleCheckChange(index)}>
            <Checkbox
              key={genre.pk}
              name={genre.name}
              id={genre.pk}
              //index={index}
              isChecked={checkedState[index]}
              //eventHandler={handleCheckChange}
            />
            </div>
```
But it results in a "Maximum updated depth exceeded" error.
Where would be the best place to combine `index` and `eventHandler`? I'm having a hard time seeing it.

Reply:
"Maximum updated depth exceeded" usually means infinite loop. In this case looks like an infinite render loop.
The new `onClick={handleCheckChange(index)}` has the parenthesis at the end, which invokes the function, which re-render the component, which invokes the function, which re-renders until you crash.

I would go for something like:
```
1 {allGenres.map((genre, index) => (
2    var changeHandler = () => {handleCheckChange(index)}
3    return <Checkbox
4        ....
5        eventHandler={changeHandler}
```

* You can add regular function lines like 2. to the arrow function on 1. as long as you denote a return on 3.
* Line 2 is similar to the handler you attached to a `div`, but doesn't invoke the handleCheckChange early, since it's wrapped in another arrow function.
* Line 5 We want to pass the handler to checkbox so it can manage when to fire the handler, but it no longer needs to know how to mix the handler with index.